### PR TITLE
Gamm Spec Out V1- Proto changes 

### PR DESCRIPTION
**Pull request by mattverse**
This PR includes proto changes to generalize gamm. WIP, and only includes the minimum changes that it requires to generalize the proto

The changes include
- pool.proto -> balancerPool.Proto
- Pool -> BalancerPool
- PoolParams -> BalancerPoolParams
- PoolAssets -> BalancerPoolAssets
- tx: MsgCreatePool -> MsgCreateBalancerPool

### CODE REVIEW

File: proto/osmosis/gamm/v1beta1/query.proto [hunk 1]
```diff
@@ -25,20 +25,23 @@ service Query {
...
   rpc TotalShares(QueryTotalSharesRequest) returns (QueryTotalSharesResponse) {
     option (google.api.http).get =
         "/osmosis/gamm/v1beta1/pools/{poolId}/total_shares";
   }
-  rpc PoolAssets(QueryPoolAssetsRequest) returns (QueryPoolAssetsResponse) {
+  rpc BalancerPoolAssets(QueryBalancerPoolAssetsRequest)
+      returns (QueryBalancerPoolAssetsResponse) {
     option (google.api.http).get =
-        "/osmosis/gamm/v1beta1/pools/{poolId}/tokens";
+        "/osmosis/gamm/v1beta1/pools/{poolId}/balancer_tokens";
```
Comment (ValarDragon) [inline proto/osmosis/gamm/v1beta1/query.proto:44]: Lets not change this

File: proto/osmosis/gamm/v1beta1/query.proto [hunk 2]
```diff
@@ -25,20 +25,23 @@ service Query {
...
+  rpc BalancerPool(QueryBalancerPoolRequest)
+      returns (QueryBalancerPoolResponse) {
     option (google.api.http).get = "/osmosis/gamm/v1beta1/pools/{poolId}";
   }
-  rpc PoolParams(QueryPoolParamsRequest) returns (QueryPoolParamsResponse) {
+  rpc BalancerPoolParams(QueryBalancerPoolParamsRequest)
+      returns (QueryBalancerPoolParamsResponse) {
     option (google.api.http).get =
-        "/osmosis/gamm/v1beta1/pools/{poolId}/params";
+        "/osmosis/gamm/v1beta1/pools/{poolId}/balancer_params";
```
Comment (ValarDragon) [inline proto/osmosis/gamm/v1beta1/query.proto:35]: Lets not change this

File: proto/osmosis/gamm/v1beta1/query.proto [hunk 3]
```diff
@@ -25,20 +25,23 @@ service Query {
   }
 
   // Per Pool gRPC Endpoints
-  rpc Pool(QueryPoolRequest) returns (QueryPoolResponse) {
+  rpc BalancerPool(QueryBalancerPoolRequest)
+      returns (QueryBalancerPoolResponse) {
     option (google.api.http).get = "/osmosis/gamm/v1beta1/pools/{poolId}";
   }
-  rpc PoolParams(QueryPoolParamsRequest) returns (QueryPoolParamsResponse) {
+  rpc BalancerPoolParams(QueryBalancerPoolParamsRequest)
```
Comment (ValarDragon) [inline proto/osmosis/gamm/v1beta1/query.proto:32]: This should still be PoolParams

File: proto/osmosis/gamm/v1beta1/query.proto [hunk 4]
```diff
@@ -25,20 +25,23 @@ service Query {
...
     option (google.api.http).get =
-        "/osmosis/gamm/v1beta1/pools/{poolId}/params";
+        "/osmosis/gamm/v1beta1/pools/{poolId}/balancer_params";
   }
   rpc TotalShares(QueryTotalSharesRequest) returns (QueryTotalSharesResponse) {
     option (google.api.http).get =
         "/osmosis/gamm/v1beta1/pools/{poolId}/total_shares";
   }
-  rpc PoolAssets(QueryPoolAssetsRequest) returns (QueryPoolAssetsResponse) {
+  rpc BalancerPoolAssets(QueryBalancerPoolAssetsRequest)
```
Comment (ValarDragon) [inline proto/osmosis/gamm/v1beta1/query.proto:41]: This should still be a query for PoolAssets

**Comment by ValarDragon** [comment]
Looks good to me except for the changes to queries.

**Comment by mattverse** [comment]
Commit "Change proto(#0a4b0f5)" removes the method GetPoolParams() from PoolI, adds the methods GetSwapFee(), GetExitFee() instead. 
Wanted to make the PR as small as possible, but it was inevitable to make this commit grouped together with this PR :(

**Comment by mattverse** [comment]
@ValarDragon Should we merge?!

**Pull request merged**